var/spack/repos/builtin/packages/py-jupyterhub/package.py

```diff
@@ -10,22 +10,21 @@ class PyJupyterhub(PythonPackage):
     """Multi-user server for Jupyter notebooks."""
 
     homepage = "https://pypi.org/project/jupyterhub"
-    url      = "https://pypi.io/packages/source/j/jupyterhub/jupyterhub-0.9.4.tar.gz"
+    url      = "https://pypi.io/packages/source/j/jupyterhub/jupyterhub-1.0.0.tar.gz"
 
-    version('0.9.4',    sha256='86b1cce446d4e8347e26913878858fc8964d103fde19b606fe37ccc5188d629d')
+    version('1.0.0',    sha256='33541a515a041b9a518ca057c1c4ab4215a7450fdddc206401713ee8137fa67f')
 
-    depends_on('python@3.5:')
-    depends_on('node-js', type=('build', 'run'))
+    depends_on('python@3.5:', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
-    depends_on('py-python-dateutil', type='run')
+    depends_on('py-python-dateutil', type=('build', 'run'))
     depends_on('py-jinja2', type='run')
-    depends_on('py-sqlalchemy', type='run')
-    depends_on('py-tornado', type='run')
-    depends_on('py-traitlets', type='run')
-    depends_on('py-alembic', type='run')
+    depends_on('py-sqlalchemy@1.1:', type=('build', 'run'))
+    depends_on('py-tornado@5.0:', type=('build', 'run'))
+    depends_on('py-traitlets@4.3.2', type=('build', 'run'))
+    depends_on('py-alembic', type=('build', 'run'))
     depends_on('py-mako', type='run')
-    depends_on('py-async-generator', type='run')
+    depends_on('py-async-generator@1.8:', type=('build', 'run'))
     depends_on('py-jupyter-notebook', type='run')
-    depends_on('py-prometheus-client', type='run')
+    depends_on('py-prometheus-client@0.0.21:', type=('build', 'run'))
     depends_on('py-send2trash', type='run')
```

Comment:
Where do these run dependencies actually come from?

Reply:
I think I found them in the documentation, but I didn't know about checking setup.py when I first created this package. If they aren't in there I can try dropping them.